### PR TITLE
chore: Remove obsolete legacy visualizations

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

@michael-s-molina apologies for all the churn, but I'm unable to re-open https://github.com/apache/superset/pull/24671 in GitHub as the branch has been forced pushed after the PR was closed.

Delving a little deeper into the annotation logic I was able to rework the logic in said PR in order for the Cypress tests to pass., i.e., this PR is a superset (no pun intended) of https://github.com/apache/superset/pull/24671 . Thankfully means that we're able to remove the `TableViz` visualization type which simplifies the visualization logic, i.e., now every chart is defined as either a legacy chart or a non-legacy chart but not both.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/common/query_context_processor.py [hunk 1]
```diff
@@ -685,22 +689,53 @@ def get_native_annotation_data(query_obj: QueryObject) -> dict[str, Any]:
...
         try:
-            viz_obj = get_viz(
-                datasource_type=chart.datasource.type,
-                datasource_id=chart.datasource.id,
-                form_data=form_data,
-                force=force,
-            )
-            payload = viz_obj.get_payload()
-            return payload["data"]
+            if chart.viz_type in viz_types:
```
Comment (john-bodley) [inline superset/common/query_context_processor.py:699]: Same logic was previously but indented.

File: superset/common/query_context_processor.py [hunk 2]
```diff
@@ -685,22 +689,53 @@ def get_native_annotation_data(query_obj: QueryObject) -> dict[str, Any]:
...
+                    _("The chart query context does not exist"),
+                )
+
+            if overrides := annotation_layer.get("overrides"):
+                if time_grain_sqla := overrides.get("time_grain_sqla"):
+                    for query_object in query_context.queries:
+                        query_object.extras["time_grain_sqla"] = time_grain_sqla
+
+                if time_range := overrides.get("time_range"):
+                    from_dttm, to_dttm = get_since_until_from_time_range(time_range)
```
Comment (john-bodley) [inline superset/common/query_context_processor.py:728]: The `QueryObject` has both `time_range` and (`from_dttm`, `to_dttm`) but I was only able to get the annotation override to adhere to the later.

File: superset/examples/birth_names.py [hunk 3]
```diff
@@ -424,6 +424,16 @@ def create_slices(tbl: SqlaTable) -> tuple[list[Slice], list[Slice]]:
                 viz_type="table",
                 metrics=metrics,
             ),
+            query_context=get_slice_json(
```
Comment (john-bodley) [inline superset/examples/birth_names.py:427]: @michael-s-molina even though you were able to run the Celery tests locally without issue, after a few tweaks (increasing timeouts to address the performance degradation on M1 silica) I was able to see the following error:

Screenshot 2023-07-14 at 8 10 46 PM

I don't fully grok where/when the query context is persisted (the only place I could see where the query context was generated from the payload/form-data was via the [/api/chart/v1/<pk>/data/](https://github.com/apache/superset/blob/0328dd270467e71260bfa85078beb7b38a87877b/superset/charts/data/api.py#L69-L172) RESTful API GET endpoint), but it does seem the existence of query context seemed somewhat sporadic. Maybe @villebro can provide more insights in terms of how/where this is generated/persisted as in the example data currently only the "Pivot Table v2" chart (per the `slices.query_context` column) has the query context explicitly defined yet there are swaths of non-legacy charts.

The TL;DR is similar to the "Pivot Table v2" chart when I explicitly added the query context for the "Daily Totals" chart (which backs the annotation layer in the Cypress test which was failing) the test passed.

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you for the cleanup.
